plugins/xml/core/src/main/java/org/pentaho/di/trans/steps/xmljoin/XMLJoinMeta.java

```diff
@@ -160,10 +161,19 @@ public void getFields( RowMetaInterface row, String name, RowMetaInterface[] inf
       // Row should only include fields from the target and not the source. During the preview table generation
       // the fields from all previous steps (source and target) are included in the row so lets remove the
       // source fields.
+      String[] targetFieldNames = null;
```

Comment:
Hi @pentaho-lmartins! To do that, I would need to change the return type of getFieldNames in RowMetaInterface.
https://github.com/pentaho/pentaho-kettle/blob/719127fa30f06dccfd3d859ea03c318da6f6ea9c/core/src/main/java/org/pentaho/di/core/row/RowMetaInterface.java#L387 

This method is heavily used on many different location. Therefore, I would say that I can't change it :P

Reply:
I mean this:

```
      List<String> targetFieldNames = null;
      RowMetaInterface targetRowMeta = transMeta.getStepFields( transMeta.findStep( getTargetXMLstep() ),
        null,
        null );
      if ( targetRowMeta != null ) {
        targetFieldNames = Arrays.asList( targetRowMeta.getFieldNames() );
      }
```

So that you don't have to create a new list for every iteration in the for cycle.

Reply:
Got it! I've updated this PR.

Reply:
Great this is OK now.